Utils/src/BTagScale.cc

```diff
@@ -0,0 +1,392 @@
+ * 
+ * Description: [one line class summary]
+ * 
+ * Implementation:
+ *     [Notes on implementation]
+ */
+//
+// Original Author:  Arne-Rasmus Draeger,68/111,4719,
+//         Created:  Fri Apr 11 16:35:33 CEST 2014
+// $Id$
+//
+//
+
+
+// system include files
+#include <memory>
+// user include files
+
+#include "CondFormats/BTauObjects/interface/BTagCalibration.h"
+#include "CondFormats/BTauObjects/interface/BTagCalibrationReader.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#
+
+#include "TFile.h"
+#include "TH1F.h"
+#include "TH2D.h"
+#include <TF1.h>
+//
+// class declaration
+//
+
+class BTagScale : public edm::EDProducer {
+public:
+  explicit BTagScale(const edm::ParameterSet&);
+  ~BTagScale();
+		
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+	
+private:
+  virtual void beginJob() ;
+  virtual void InitTagEff(std::string fname);
+  virtual void FillEffSF(const pat::Jet &aJet, double&Eff, double&SF,double&SFUp, double&SFDown);
+  virtual void produce(edm::Event&, const edm::EventSetup&);
+  virtual void endJob() ;
+	
+  virtual void beginRun(edm::Run&, edm::EventSetup const&);
+  virtual void endRun(edm::Run&, edm::EventSetup const&);
+  virtual void beginLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+  virtual void endLuminosityBlock(edm::LuminosityBlock&, edm::EventSetup const&);
+  edm::InputTag JetTag_;
+  std::string   btagEffFile_;
+  std::string   CSVSFFile_;
+  //fill these based on the Mass Point (Mg,mLSP)
+  TH2D*btagEff;
+  TH2D*ctagEff;
+  TH2D*ltagEff;
+	 	
+	
+  // ----------member data ---------------------------
+};
+
+//
+// constants, enums and typedefs
+//
+
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+BTagScale::BTagScale(const edm::ParameterSet& iConfig)
+{
+  //register your produc
+
+  JetTag_ = iConfig.getParameter<edm::InputTag>("JetsTag");
+  btagEffFile_=iConfig.getParameter<std::string>  ("BTagEffInput");
+  CSVSFFile_=iConfig.getParameter<std::string>("CSVTag");
+  InitTagEff(btagEffFile_);
+  //now do what ever other initialization is needed
+  produces<double>("BTagProb0");	
+  produces<double>("BTagProb1");
+  produces<double>("BTagProb2");
+  produces<double>("BTagProb3");
+
+  produces<double>("BTagProb0ShiftUp");
+  produces<double>("BTagProb1ShiftUp");
+  produces<double>("BTagProb2ShiftUp");
+  produces<double>("BTagProb3ShiftUp");	
+
+  produces<double>("BTagProb0ShiftDown");
+  produces<double>("BTagProb1ShiftDown");
+  produces<double>("BTagProb2ShiftDown");
+  produces<double>("BTagProb3ShiftDown");
+
```

Comment:
Sure I don’t mind. I made them doubles because it made it easier for plotting the nominal and shifted b-tag shapes. 
> On Oct 5, 2015, at 5:45 PM, Kevin Pedro <notifications@github.com> wrote:
> 
> In Utils/src/BTagScale.cc <https://github.com/TreeMaker/TreeMaker/pull/68#discussion_r41159839>:
> 
> > +  //now do what ever other initialization is needed
> > +  produces<double>("BTagProb0");	
> > +  produces<double>("BTagProb1");
> > +  produces<double>("BTagProb2");
> > +  produces<double>("BTagProb3");
> > +
> > +  produces<double>("BTagProb0ShiftUp");
> > +  produces<double>("BTagProb1ShiftUp");
> > +  produces<double>("BTagProb2ShiftUp");
> > +  produces<double>("BTagProb3ShiftUp");	
> > +
> > +  produces<double>("BTagProb0ShiftDown");
> > +  produces<double>("BTagProb1ShiftDown");
> > +  produces<double>("BTagProb2ShiftDown");
> > +  produces<double>("BTagProb3ShiftDown");
> > +
> Could these products be 3 vectors instead of 12 doubles?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/TreeMaker/TreeMaker/pull/68/files#r41159839>.
>